components/style/properties/longhand/ui.mako.rs

```diff
@@ -29,3 +29,38 @@ ${helpers.single_keyword("-moz-window-dragging", "default drag no-drag", product
                          gecko_enum_prefix="StyleWindowDragging",
                          animatable=False,
                          spec="None (Nonstandard Firefox-only property)")}
+
+<%helpers:longhand name="caret-color" animatable="False" products="none"
+                   spec="https://drafts.csswg.org/css-ui/#caret-color">
+    use std::fmt;
+    use style_traits::ToCss;
+    pub use self::computed_value::T as SpecifiedValue;
+
+    pub mod computed_value {
+        use cssparser::{Color, Parser};
+        use parser::{Parse, ParserContext};
+        use values::{Auto, Either};
+        use values::HasViewportPercentage;
+        use values::computed::ComputedValueAsSpecified;
+
+        no_viewport_percentage!(Color);
+        impl ComputedValueAsSpecified for Color {}
+
+        impl Parse for Color {
```

Comment:
@Wafflespeanut that worked with a definition for caret-color as follows

```
${helpers.predefined_type("caret-color", 
                          "ColorOrAuto", 
                          "Either::Second(Auto)",
                          spec="https://drafts.csswg.org/css-ui/#caret-color",
                          animatable="False",
                          boxed="True")}
```

I also had to add that type definition in both `values/specified/mod.rs` and `values/computed/mod.rs`. Is that how we want to go about implementing this? If so I can push up those changes.